### DISCUSSION TITLE
Archiving Workflow Module - recover from failed archived workflows

### DISCUSSION THREAD

**Discussion by YoBeMo** [Netflix/conductor#3031] [General]
Hi,

In our setup we use the com.netflix.conductor.contribs.ArchivingWorkflowModule and ES7 for indexing.
In some cases a workflow might fail to archive - network issues, cluster state, etc... 
Is there any existing mechanism that handles this?

We use Redis as persistent layer and cases like this result in entries without TTL in Redis that would never be evicted.

Any thoughts about this?

**Comment by aravindanr** [comment]
From the looks of it, `com.netflix.conductor.contribs.ArchivingWorkflowModule` is part of v2.31.x which is no longer supported. The `ArchivingWorkflowModule` sets up the [ArchivingWorkflowStatusListener](https://github.com/Netflix/conductor/blob/2.31/contribs/src/main/java/com/netflix/conductor/contribs/listener/ArchivingWorkflowStatusListener.java) which is now part of  [conductor-community](https://github.com/Netflix/conductor-community/tree/main/workflow-event-listener/src/main/java/com/netflix/conductor/contribs/listener/archive).

if the persistence layer is Redis, retries can be configured in the Redis client used by the `RedisExecutionDAO`. We highly recommend upgrading to the latest stable version, so we can better support you.

**Reply by karanbhola** [reply]
how can we use the ArchivingWorkflowModule inside the conductor or can you provide an example on how we can use the archival of workflow executions?
